stackslib/src/chainstate/nakamoto/staging_blocks.rs

```diff
@@ -0,0 +1,396 @@
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::fs;
+use std::ops::{Deref, DerefMut};
+use std::path::PathBuf;
+
+use lazy_static::lazy_static;
+use rusqlite::types::{FromSql, FromSqlError};
+use rusqlite::{params, Connection, OpenFlags, OptionalExtension, ToSql, NO_PARAMS};
+use stacks_common::types::chainstate::{ConsensusHash, StacksBlockId};
+use stacks_common::util::{get_epoch_time_secs, sleep_ms};
+
+use crate::chainstate::burn::db::sortdb::{SortitionDB, SortitionHandle};
+use crate::chainstate::burn::BlockSnapshot;
+use crate::chainstate::nakamoto::{NakamotoBlock, NakamotoChainState};
+use crate::chainstate::stacks::db::StacksChainState;
+use crate::chainstate::stacks::index::marf::MarfConnection;
+use crate::chainstate::stacks::{Error as ChainstateError, StacksBlock, StacksBlockHeader};
+use crate::stacks_common::codec::StacksMessageCodec;
+use crate::util_lib::db::{
+    query_int, query_row, query_row_panic, query_rows, sqlite_open, tx_begin_immediate, u64_to_sql,
+    DBConn, Error as DBError, FromRow,
+};
+
+lazy_static! {
+    pub static ref NAKAMOTO_STAGING_DB_SCHEMA_1: Vec<String> = vec![
+      r#"
+      -- Table for staging nakamoto blocks
+      CREATE TABLE nakamoto_staging_blocks (
+                     -- SHA512/256 hash of this block
+                     block_hash TEXT NOT NULL,
+                     -- the consensus hash of the burnchain block that selected this block's miner's block-commit
+                     consensus_hash TEXT NOT NULL,
+                     -- the parent index_block_hash
+                     parent_block_id TEXT NOT NULL,
+
+                     -- has the burnchain block with this block's `consensus_hash` been processed?
+                     burn_attachable INT NOT NULL,
+                     -- has this block been processed?
+                     processed INT NOT NULL,
+                     -- set to 1 if this block can never be attached
+                     orphaned INT NOT NULL,
+
+                     height INT NOT NULL,
+
+                     -- used internally -- this is the StacksBlockId of this block's consensus hash and block hash
+                     index_block_hash TEXT NOT NULL,
+                     -- how long the block was in-flight
+                     download_time INT NOT NULL,
+                     -- when this block was stored
+                     arrival_time INT NOT NULL,
+                     -- when this block was processed
+                     processed_time INT NOT NULL,
+
+                     -- block data
+                     data BLOB NOT NULL,
+                    
+                     PRIMARY KEY(block_hash,consensus_hash)
+        );"#
+        .into(),
+        r#"CREATE INDEX by_index_block_hash ON nakamoto_staging_blocks(index_block_hash);"#.into()
+    ];
+}
```

Comment:
`Vec::new` and `vec![]` are `const` functions -- meaning they can be invoked on `const` arguments and used in `const` definitions without `lazy_static!`. This hasn't always been a feature, which is partially why we do still have so many uses of `lazy_static!`, but when its easy to avoid, I think we should try to do so:


```suggestion
pub const NAKAMOTO_STAGING_DB_SCHEMA_1: Vec<&'static str> = vec![
      r#"
      -- Table for staging nakamoto blocks
      CREATE TABLE nakamoto_staging_blocks (
                     -- SHA512/256 hash of this block
                     block_hash TEXT NOT NULL,
                     -- the consensus hash of the burnchain block that selected this block's miner's block-commit
                     consensus_hash TEXT NOT NULL,
                     -- the parent index_block_hash
                     parent_block_id TEXT NOT NULL,

                     -- has the burnchain block with this block's `consensus_hash` been processed?
                     burn_attachable INT NOT NULL,
                     -- has this block been processed?
                     processed INT NOT NULL,
                     -- set to 1 if this block can never be attached
                     orphaned INT NOT NULL,

                     height INT NOT NULL,

                     -- used internally -- this is the StacksBlockId of this block's consensus hash and block hash
                     index_block_hash TEXT NOT NULL,
                     -- how long the block was in-flight
                     download_time INT NOT NULL,
                     -- when this block was stored
                     arrival_time INT NOT NULL,
                     -- when this block was processed
                     processed_time INT NOT NULL,

                     -- block data
                     data BLOB NOT NULL,
                    
                     PRIMARY KEY(block_hash,consensus_hash)
        );"#,
        r#"CREATE INDEX by_index_block_hash ON nakamoto_staging_blocks(index_block_hash);"#,
];
```

Reply:
Done